curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-data-structures-by-building-a-shopping-cart/63f02a4ef92d711ec1ff618c.md

```diff
@@ -0,0 +1,375 @@
+---
+id: 63f02a4ef92d711ec1ff618c
+title: Step 43
+challengeType: 0
+dashedName: step-43
+---
+
+# --description--
+
+You also need to update the total price of the cart when the user adds an item. Create a `calculateTotal` method in the `ShoppingCart` class.
+
+In that method, declare a `subTotal` variable and use the `reduce` method on the `items` array to calculate the sum of the `price` property of each item in the array. Remember to set your initial value in the `reduce` method.
+
+# --hints--
+
+You should create a `calculateTotal` method in the `ShoppingCart` class.
+
+```js
+assert.isFunction(cart.calculateTotal);
+```
+
+Your `calculateTotal` method should have a `subTotal` variable declared with `const`.
+
+```js
+const afterCalculateTotal = code.split('calculateTotal')[1];
+assert.match(afterCalculateTotal, /const\s+subTotal\s*=/);
+```
+
+Your `calculateTotal` method should use the `reduce` method on the `items` array.
+
+```js
+const afterCalculateTotal = code.split('calculateTotal')[1];
+assert.match(afterCalculateTotal, /this\s*\.\s*items\s*\.\s*reduce/);
+```
+
+Your `reduce` callback should use `total` and `item` as the first and second parameter. Remember to use arrow function syntax.
```

Comment:
We should add this to the instructions because it is not clear that campers have to use these parameter names